lib/tasks/maintenance.rake

```diff
@@ -33,4 +33,21 @@ namespace :maintenance do
     end
     p "cleared old discarded api_actions @ #{Time.now}"
   end
+
+  task :purge_old_api_resources => [:environment] do 
+    Subdomain.all_with_public_schema.each do |subdomain|
+      Apartment::Tenant.switch subdomain.name do
+        ApiNamespace.where.not(purge_resources_older_than: ApiNamespace::RESOURCES_PURGE_INTERVAL_MAPPING[:never]).each do |api_namespace|
```

Comment:
lets read in batches, there is a huge number of records

https://api.rubyonrails.org/classes/ActiveRecord/Batches.html

Reply:
@Pralish awaiting this change before merging ⏫

Reply:
@donrestarone , This line reads the api_namespaces for a single subdomain. 

Here's the method that will read and destroy api_resources in batches.
https://github.com/restarone/violet_rails/pull/1492/files#diff-03241a9bad7359398fa0da5610d4fd762f027297ee8acb1326c900d1d8fe49f1R397-R399

Reply:
yes, but we should use the same method for performing the destruction in batches. 

The problem right now is that we delete in 2 different ways. The rake task can overload a system from the background if it reads too much data.

Reply:
also, we should read in batches here `ApiNamespace.where.not(purge_resources_older_than: ApiNamespace::RESOURCES_PURGE_INTERVAL_MAPPING[:never]).each`

Reply:
actually @Pralish Im seeing it now-- we don't have to read the namespaces in batches. This is fine ✅